GetOldTweets3/manager/ConcurrentTweetManager.py

```diff
@@ -0,0 +1,78 @@
+# Author: Giulio Neusch-Frediani - www.github.com/giulionf
+from datetime import datetime
+from threading import Thread
+from queue import Queue
+from queue import Empty
+import copy
+
+from GetOldTweets3.manager.TweetManager import TweetManager
+
+DATE_FORMAT = "%Y-%m-%d"
+FIRST_TWEET_DATE = datetime.strptime("2006-03-21", DATE_FORMAT)
+
+
+class ConcurrentTweetManager:
+
+    @staticmethod
+    def getTweets(tweetCriteria, receiveBuffer=None, bufferLength=100, proxy=None, debug=False, worker_count=1, ordered=True):
+        # Init the queues
+        time_spans = Queue()
+        tweets = []
+        workers = []
+
+        # Split the date in smaller parts
+        since_date = datetime.strptime(tweetCriteria.since, DATE_FORMAT) if tweetCriteria.since else FIRST_TWEET_DATE
+        until_date = datetime.strptime(tweetCriteria.until, DATE_FORMAT) if tweetCriteria.until else datetime.now()
+        date_diff_per_worker = (until_date - since_date) / worker_count
+
+        # Create a TweetCriteria that can be cloned as Model by the Workers
+        criteria = copy.deepcopy(tweetCriteria)
+        criteria.setMaxTweets(criteria.maxTweets / worker_count)
+
+        # Create a time span for each of the splitted parts and a corresponding worker
+        for i in range(1, worker_count+1):
+            from_time = since_date + (i-1) * date_diff_per_worker
+            to_time = since_date + i * date_diff_per_worker
+            time_spans.put((from_time, to_time))
+
+            w = WorkerThread(copy.deepcopy(criteria), time_spans, tweets, receiveBuffer, bufferLength, proxy, debug)
```

Comment:
What is the purpose of making a copy of the copy by `copy.deepcopy(criteria)`?

Reply:
In the worker thread i need to set the from_date and until_date in the criteria. If i do not do a deep copy there will be a race condition for the criteria between the threads.

Reply:
Ok